File: dagger2demo/src/main/java/com/example/dagger2demo/BaseApplication.java
package com.example.dagger2demo;

import android.app.Application;
import com.example.dagger2demo.component.DaggerMyComponent;
import com.example.dagger2demo.component.MyComponent;
import com.example.dagger2demo.module.DatabaseModule;
import com.example.dagger2demo.module.HttpModule;

/**
 * Created by zhengwei on 2021/5/23.
 */
public class BaseApplication extends Application {

    private MyComponent myComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        myComponent = DaggerMyComponent.builder()
                .httpModule(new HttpModule())
                .databaseModule(new DatabaseModule())
                .build();
    }

    public MyComponent getAppMyComponent() {
        return myComponent;
    }
}

File: hiltdemo/src/main/java/com/example/hiltdemo/BaseApplication.kt
package com.example.hiltdemo

import android.app.Application
import dagger.hilt.android.HiltAndroidApp
import javax.inject.Inject

/**
 * Created by zhengwei on 2021/5/23.
 */
@HiltAndroidApp
class BaseApplication : Application() {

    @Inject
    lateinit var hiltTest: HiltTest

    override fun onCreate() {
        super.onCreate()
        hiltTest.hiltTest()
    }
}